app/serializers/school_serializer.rb

```diff
@@ -3,47 +3,36 @@ class AttributesSerializer < Blueprinter::Base
     exclude :id
 
     field :name
-    field(:urn) { |school, _| school.urn.to_s }
-    field(:cohort) do |school, options|
-      school_contract_period_year(school, options)
+    field(:urn) do |school, _|
+      school.urn.to_s
+    end
+    field(:cohort) do |_, options|
+      options[:contract_period_year].to_s
     end
     field(:in_partnership) do |school, options|
-      in_partnership?(school, options)
+      contract_period_metadata(school:, options:).in_partnership
     end
     field(:induction_programme_choice) do |school, options|
-      school.training_programme_for(school_contract_period_year(school, options))
+      contract_period_metadata(school:, options:).induction_programme_choice
     end
     field(:expression_of_interest) do |school, options|
-      expressions_of_interest?(school, options)
+      lead_provider_contract_period_metadata(school:, options:).expression_of_interest
     end
     field :created_at
     field(:api_updated_at, name: :updated_at)
 
     class << self
-      def school_contract_period_year(school, options)
-        if school.respond_to?(:transient_contract_period_year)
-          school.transient_contract_period_year
-        else
-          options[:contract_period_year].to_s
-        end
-      end
-
-      def in_partnership?(school, options)
-        if school.respond_to?(:transient_in_partnership)
-          school.transient_in_partnership
-        else
-          school.school_partnerships.for_contract_period(options[:contract_period_year]).exists?
-        end
+      def contract_period_metadata(school:, options:)
+        school.contract_period_metadata.select(
```

Comment:
I'm getting an `ActiveRecord::SoleRecordExceeded` error here for some reason.. I've changed these 2 methods as per below to fix it:

```
def contract_period_metadata(school:, options:)
  school.contract_period_metadata.select { |cpmt|
    cpmt.contract_period_year = options[:contract_period_year]
  }.sole
end

def lead_provider_contract_period_metadata(school:, options:)
  school.lead_provider_contract_period_metadata.select { |cpmt|
    cpmt.lead_provider_id = options[:lead_provider_id] &&
      cpmt.contract_period_year = options[:contract_period_year]
  }.sole
end
```